cli/command/image/tree.go

```diff
@@ -59,6 +59,11 @@ func runTree(ctx context.Context, dockerCLI command.Cli, opts treeOptions) error
 				},
 			}
 
+			if sub.Details.Used {
+				// Mark top-level parent image as used if any of its subimages are used.
+				details.Used = true
```

Comment:
I forgot now; did we also add the list of container-id's that used it somewhere? (Maybe not to be printed by default for sure)

Reply:
Yes, the `ImageData` has a list of containers. It's only available in the manifest summary though.
Also, see: https://github.com/docker/cli/pull/4982#discussion_r1553521166